code/obj/item/kitchen.dm

```diff
@@ -731,6 +742,94 @@ TRAYS
 		if (ishuman(AM) && AM.throwing) // only humans have the power to smash plates with their bodies
 			src.shatter()
 
+/obj/item/plate/pizza_box
+	name = "pizza box"
+	desc = "Can hold wedding rings, clothes, weaponry... and sometimes pizza."
+	icon = 'icons/obj/large_storage.dmi'
+	icon_state = "pizzabox" // to avoid using plate SFX
+	pickup_sfx = 0
+	w_class = W_CLASS_BULKY
+	inhand_image_icon = 'icons/mob/inhand/hand_food.dmi'
+	item_state = "pizza_box"
+	is_plate = FALSE
+	var/open = FALSE
+
+	add_contents(obj/item/food, mob/user, click_params) // Due to non-plates skipping some checks in the original add_contents() we'll have to do our own checks.
+
+		if (!src.open && !istype(food, /obj/item/plate/))
+			boutput(user, "<span class='alert'>You have to open \the [src] to put something in it, silly!</span>")
+			return
+
+		if (src.open && istype(food, /obj/item/plate/))
+			boutput(user, "<span class='alert'>You can only put \the [food] on top of \the [src] when it's closed!")
+			return
+
+		if (length(src.foods_inside) == src.max_food && !istype(food, src.type))
+			boutput(user, "You try to think of a way to put [food] in \the [src] but it's not possible! It's too large!")
```

Comment:
shouldn't this say it's full, not that the item is too large?

Reply:
solved in my other commit.